### PR TITLE
fix fit_generator for tfdatasets

### PR DESCRIPTION
**Pull request by skeydan**
not yet a fix for #682 but I something I noticed on the way... adding corresponding test in `tfdatasets`.

### CODE REVIEW

File: R/model.R [hunk 1]
```diff
@@ -925,11 +925,14 @@ as_generator.default <- function(x) {
 }
 
 as_generator.tensorflow.python.data.ops.dataset_ops.Dataset <- function(x) {
-  python_path <- system.file("python", package = "tfdatasets")
+  python_path <- system.file("python", package = "keras")
```
Comment (jjallaire) [inline R/model.R]: That was a big oversight! Glad you caught that one.

Comment (skeydan) [reply R/model.R]: yeah well that only resulted in a warning from `normalizePath`, but it still worked until 1.13, but for 1.13 we need the `DatasetV2` thing...

**Comment by skeydan** [comment]
This is interesting!!! Because of the `keras`-related tests failing I thought we'd finally have to add "duplicate" S3 implementations for `tensorflow.python.keras`.
But what makes the difference is whether I `require(keras)` inside a call to `test_succeeds` (as it was done previously) or directly at the top level of the file. (I remember last time I tried requiring it in `testthat.R`, which did not yield the desired results...).

So at least for this reason (`tfdatasets` tests) we don't have to add those implementations. 
Final (hopefully) commit for this following soon...

**Comment by skeydan** [comment]
getting rid of the hack I introduced for the preexisting `keras`-related test now ...

>     # Hack due to the keras onload hook that replaces "tensorflow.python.keras" in
    # class names by "keras" not being executed when the test is run.
    # Remove in case we later decide to provide S3 implementations for
    # tensorflow.python.keras.xxx.

**Comment by skeydan** [comment]
Sorry, the above 2 comments should have gone here: https://github.com/rstudio/tfdatasets/pull/28 (why did I have to put `tfdatasets` in the title???)
